doc/ni_by_example.md

```diff
@@ -0,0 +1,564 @@
+
+####`F`: Split text stream into columns
+
+* `F:<char>`: split on character
+  * Note: some characters need to be escaped (for example, forward-slash); use `F/regex/` below for more flexibility (at the cost of less concision).
+* `F/regex/`: split on occurrences of regex. If present, the first capture group will be included before a tab is appended to a field.
+* `Fm/regex/`: don't split; instead, look for matches of regex and use those as the field values.
+* `FC`: split on commas (doesn't handle special CSV cases)
+* `FV`: parse CSV "correctly," up to newlines in fields
+* `FS`: split on runs of horizontal whitespace
+* `FW`: split on runs of non-word characters
+* `FP`: split on pipe symbols
+
+####`p'F_ ...'; p'FM'; p'FR n'`: Explicit field access
+
+In general, `ni` data will be long and narrow--that is, it will have millions to trillions of rows in the stream, but usually no more than a dozen relevant features per row.
+
+However, `ni` implements access to fields beyond the first 12 using the explicit field accessors `p'F_ ...', p'FM', p'FR n'`
+
+It has not occurred in my experience that I have needed to maintain more than 12 relevant columns, and as a result I find the syntax a bit hard to remember, because I think of `A` as the first letter, rather than the zeroth, which is how `ni` thinks, internally.
+
+`p'F_ ...'` takes a range (or a single number) as its second argument.
+
+To print fields 11-15 of a data source, you would use `$ ni ... r F_ 10..14`.
+
+The index of the final nonempty field in a line is stored in `FM`. To get the last field of every line in our example, you could use the spell: `$ ni /usr/share/dict/words rx40 r10 p'r substr(a, 0, 3), substr(a, 3, 3), substr(a, 6)' p'r FM, F_ FM'`
+
+It is often useful to take everything after a certain point in a line. This can be accomplished efficiently using the `FR n` operator. `FR n` is equivalent to `F_ n..FM`.
+
+
+
+##Sort, Unique, and Count
+Sorting large amounts of data requires buffering to disk, and the vagaries of how this works internally is a source of headaches and slow performance. If your data is larger than a gigabyte uncompressed, you may want to take advantage of massively distributing the workload through Hadoop operations.
+
+If you're not convinced that anything could go slow in `ni`, let's try counting all of the letters in the dictionary 
+
+`$ ni /usr/share/dict/words F// p'FR 0' gc \>letter_counts.txt`
+
+You'll probably see the `ni` monitor for the first time, showing the steps of the computation, and what steps are taking time.  The whole process takes about 90 seconds on my computer.
+
+```
+$ ni --explain /usr/share/dict/words F// p'FR 0' gc \>letter_counts.txt
```

Comment:
`p'FR 0'` is probably more idiomatic as `pF_` -- though I don't remember whether you've introduced `F_` yet